File: C++/src/Solution2801_2900/Solution2810.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution2810.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string finalString(string s) {
  string res = "";
  for (const auto &item : s){
    if (item == 'i'){
      std::reverse(res.begin(), res.end());
    }else{
      res.push_back(item);
    }
  }
  return res;
}

File: Java/src/main/java/QuestionBank/Solution2801_2900/Solution2810.java
package QuestionBank.Solution2801_2900;
/**
 * @author windows
 */
public class Solution2810 {

    public String finalString(String s) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'i'){
                sb.reverse();
            }else {
                sb.append(s.charAt(i));
            }
        }
        return sb.toString();
    }
    
    
    public static void main(String[] args) {
        Solution2810 solution2810 = new Solution2810();
    }
}